File: app/src/main/java/com/example/lun/pocket_health_advisor/videoCall/SinchLoginActivity.java
package com.example.lun.pocket_health_advisor.videoCall;

import android.app.ProgressDialog;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.firebase.auth.FirebaseAuth;
import com.sinch.android.rtc.SinchError;

public class SinchLoginActivity extends BaseActivity implements SinchService.StartFailedListener {

    private Button mLoginButton;
    private EditText mLoginName;
    private ProgressDialog mSpinner;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_sinch_login);
//
//        mLoginName = (EditText) findViewById(R.id.loginName);
//
//        mLoginButton = (Button) findViewById(R.id.loginButton);
//        mLoginButton.setEnabled(false);
//        mLoginButton.setOnClickListener(new OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                loginClicked();
//            }
//        });
    }

    @Override
    protected void onServiceConnected() {
        //mLoginButton.setEnabled(true);
        getSinchServiceInterface().setStartListener(this);
        if (FirebaseAuth.getInstance().getCurrentUser() != null) {
//            loginClicked(FirebaseAuth.getInstance().getCurrentUser().getDisplayName());
            loginClicked("111");
        }
    }

    @Override
    protected void onPause() {
        if (mSpinner != null) {
            mSpinner.dismiss();
        }
        super.onPause();
    }

    @Override
    public void onStartFailed(SinchError error) {
        Toast.makeText(this, error.toString(), Toast.LENGTH_LONG).show();
        if (mSpinner != null) {
            mSpinner.dismiss();
        }
    }

    @Override
    public void onStarted() {
        openPlaceCallActivity();
    }

    private void loginClicked(String userName) {

//        if (userName.isEmpty()) {
//            Toast.makeText(this, "Please enter a hospitalName", Toast.LENGTH_LONG).show();
//            return;
//        }

        if (!userName.equals(getSinchServiceInterface().getUserName())) {
            getSinchServiceInterface().stopClient();
        }

        if (!getSinchServiceInterface().isStarted()) {
            getSinchServiceInterface().startClient(userName);
            showSpinner();
        } else {
            openPlaceCallActivity();
        }
    }

    private void openPlaceCallActivity() {
        finish();
//        Intent mainActivity = new Intent(this, PlaceCallActivity.class);
//        startActivity(mainActivity);
    }

    private void showSpinner() {
        mSpinner = new ProgressDialog(this);
        mSpinner.setTitle("Logging in");
        mSpinner.setMessage("Please wait...");
        mSpinner.show();
    }
}

File: app/src/main/java/com/example/lun/pocket_health_advisor/SinchLoginActivity.kt
package com.example.lun.pocket_health_advisor


import com.bra

import android.app.ProgressDialog
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.View.OnClickListener
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class LoginActivity : BaseActivity(), SinchService.StartFailedListener {

    private var mLoginButton: Button? = null
    private var mLoginName: EditText? = null
    private var mSpinner: ProgressDialog? = null

    protected fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.login)

        mLoginName = findViewById(R.id.loginName) as EditText

        mLoginButton = findViewById(R.id.loginButton) as Button
        mLoginButton!!.isEnabled = false
        mLoginButton!!.setOnClickListener { loginClicked() }
    }

    protected fun onServiceConnected() {
        mLoginButton!!.isEnabled = true
        getSinchServiceInterface().setStartListener(this)
    }

    protected fun onPause() {
        if (mSpinner != null) {
            mSpinner!!.dismiss()
        }
        super.onPause()
    }

    fun onStartFailed(error: SinchError) {
        Toast.makeText(this, error.toString(), Toast.LENGTH_LONG).show()
        if (mSpinner != null) {
            mSpinner!!.dismiss()
        }
    }

    fun onStarted() {
        openPlaceCallActivity()
    }

    private fun loginClicked() {
        val userName = mLoginName!!.text.toString()

        if (userName.isEmpty()) {
            Toast.makeText(this, "Please enter a name", Toast.LENGTH_LONG).show()
            return
        }

        if (userName != getSinchServiceInterface().getUserName()) {
            getSinchServiceInterface().stopClient()
        }

        if (!getSinchServiceInterface().isStarted()) {
            getSinchServiceInterface().startClient(userName)
            showSpinner()
        } else {
            openPlaceCallActivity()
        }
    }

    private fun openPlaceCallActivity() {
        val mainActivity = Intent(this, PlaceCallActivity::class.java)
        startActivity(mainActivity)
    }

    private fun showSpinner() {
        mSpinner = ProgressDialog(this)
        mSpinner!!.setTitle("Logging in")
        mSpinner!!.setMessage("Please wait...")
        mSpinner!!.show()
    }
}
